# DATA PREPARATION

## Exploring `DataFrame` informations

> **Learning goals:** By the end of this subsections, you'll be comfortable for finding general information in a `DataFrame`.

We will import the Python scikit-learn library and use an iconic dataset that every data scientist has seen hundreds of times: British biologist Ronald Fisher's *Iris* data set used in his 1936 paper "The use of multiple measurements in taxonomic problems":

In [5]:
import sys
!{sys.executable} -m pip scikit-learn

ERROR: unknown command "scikit-learn"



In [9]:
import pandas as pd
from sklearn.datasets import load_iris

iris = load_iris()
iris_df = pd.DataFrame(data=iris['data'], columns=iris['feature_names'])




### `DataFrame.shape`

Knowing the shape of the `DataFrame` is super-important because it revealed the volume of the dataset you are working with. To explore the shape of a `DataFrame` you can use `DataFrame.shape` method

```python
df.shape
```

In [10]:
iris_df.shape

(150, 4)

Iris dataset has 150 rows and 4 columns

### `DataFrame.columns`

Since the Iris dataset has 4 columns, we can dive to know the title of the columns using `DataFrame.columns`.
```python
df.columns
```

In [12]:
iris_df.columns

Index(['sepal length (cm)', 'sepal width (cm)', 'petal length (cm)',
       'petal width (cm)'],
      dtype='object')

### `DataFrame.info`

`DataFrame.info()` gives more features than the combine of the above methods. It gives details of the dataset including: 
columns name, missing data, data type and memory usage


In [13]:
iris_df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 150 entries, 0 to 149
Data columns (total 4 columns):
 #   Column             Non-Null Count  Dtype  
---  ------             --------------  -----  
 0   sepal length (cm)  150 non-null    float64
 1   sepal width (cm)   150 non-null    float64
 2   petal length (cm)  150 non-null    float64
 3   petal width (cm)   150 non-null    float64
dtypes: float64(4)
memory usage: 4.8 KB
